### PR TITLE
Added custom parameter types for cucubmer js

### PR DESCRIPTION
**Pull request by KirillBogdanets**

### CODE REVIEW

File: src/parameterTypes.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+      ArrowRight: "ArrowRight",
+      ArrowLeft: "ArrowLeft",
+    };
+    return KEYS[keyName];
+  },
+  name: "key",
+  useForSnippets: false,
+});
+
+/**
```
Comment (nomid0612) [inline src/parameterTypes.ts:29]: pls change description, seems like it is for waits

File: src/parameterTypes.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+      "be focused": "isFocused",
+      "be displayed in viewport": "isDisplayedInViewport",
+    };
+    return CONDITIONS[condition];
+  },
+  name: "expect condition",
+  useForSnippets: false,
+});
+
+/**
```
Comment (nomid0612) [inline src/parameterTypes.ts]: description is not for chai

File: src/parameterTypes.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+      exist: "waitForExist",
+      "be enabled": "waitForEnabled",
+      "be displayed": "waitForDisplayed",
+      "be clickable": "waitForClickable",
+      "be focused": "waitForElementFocused",
+      "be displayed in viewport": "waitForElementDisplayedInViewport",
+    };
+    return CONDITIONS[condition];
+  },
+  name: "wait condition",
```
Comment (AlexGalichenko) [inline src/parameterTypes.ts]: As agreed. Let's change to camel case.

File: src/parameterTypes.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    return CONDITIONS[condition];
+  },
+  name: "expect condition",
+  useForSnippets: false,
+});
+
+/**
+ * Used for returning chai assertion condition
+ */
+defineParameterType({
```
Comment (AlexGalichenko) [inline src/parameterTypes.ts:32]: I've already implemented generic way to resolve negation in validation type.
```
export function validationTransformer(p: string) {
    const regexp: RegExp = /(does not )?(.+)s?/;
    const [ _, reverse, validation ] = p.match(regexp) as RegExpMatchArray;
    return function (AR: any, ER: any) {
        verify({ AR, ER, validation, reverse: Boolean(reverse) })
    }
}
```

File: src/parameterTypes.ts [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    return !!condition;
+  },
+  name: "case sensitive",
+  useForSnippets: false,
+});
+
+/**
+ * Used to compare numbers
+ */
+defineParameterType({
```
Comment (AlexGalichenko) [inline src/parameterTypes.ts:56]: maybe it's better to extend generic verify function adding these belov/above conditions?

File: src/parameterTypes.ts [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+  transformer(condition: string) {
+    const CONDITIONS: any = {
+      exist: "waitForExist",
+      "be enabled": "waitForEnabled",
+      "be displayed": "waitForDisplayed",
+      "be clickable": "waitForClickable",
+      "be focused": "waitForElementFocused",
+      "be displayed in viewport": "waitForElementDisplayedInViewport",
+    };
+    return CONDITIONS[condition];
```
Comment (AlexGalichenko) [inline src/parameterTypes.ts]: what about to add negation to this param type and return function that performs certain wait.

File: src/parameterTypes.ts (hunk 1 above)
Comment (KirillBogdanets) [reply src/parameterTypes.ts:29]: done

File: src/parameterTypes.ts (hunk 2 above)
Comment (KirillBogdanets) [reply src/parameterTypes.ts]: done

File: src/parameterTypes.ts (hunk 3 above)
Comment (KirillBogdanets) [reply src/parameterTypes.ts]: removed

File: src/parameterTypes.ts (hunk 4 above)
Comment (KirillBogdanets) [reply src/parameterTypes.ts:32]: removed

File: src/parameterTypes.ts (hunk 6 above)
Comment (KirillBogdanets) [reply src/parameterTypes.ts]: removed

File: package.json [hunk 7]
```diff
@@ -19,16 +26,33 @@
...
+    "eslint-plugin-wdio": "^7.4.2",
+    "husky": "^4.3.8",
     "jest": "^28.1.1",
+    "prettier": "^2.3.2",
     "ts-jest": "^28.0.5",
     "typescript": "^4.7.3"
   },
   "dependencies": {
+    "@cucumber/cucumber": "^8.2.2",
+    "@types/chai": "^4.3.1",
```
Comment (AlexGalichenko) [inline package.json]: plz, move types/chai to dev dependency

**Review by AlexGalichenko — APPROVED** [review]

**Pull request merged**